tensordict/tensordict.py

```diff
@@ -1791,6 +1800,10 @@ def __iter__(self) -> Generator:
     def flatten_keys(
         self, separator: str = ".", inplace: bool = False
     ) -> TensorDictBase:
+        if self._being_flattened:
```

Comment:
One idea could be to decorate this with a `try / except / finally` to make sure that this is fault-tolerant:
```
try:
    ...
    self._being_flattened = True
    ...
finally:
    self._being_flattened = False
```

Reply:
Got it. Make sense